### DISCUSSION TITLE
Welcome to acme.sh Discussions!

### DISCUSSION THREAD

**Discussion by neilpang** [acmesh-official/acme.sh#4258] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Eddict** [comment]
hi there,
i would like to implement a new DNS service based on the [DirectAdmin API](https://www.directadmin.com/api.php) and its CMD_API_DNS_CONTROL interface.
what are the requirements (eg. public function(name)s, logging, error trapping, etc) for a (new) service/script to be able to work with the acme.sh shell script? is there any description of that? it seems like _add() and _rm() are implemented in all of them..
regards Eddict

**Reply by Eddict** [reply]
my bad, found it 🤦🏽‍♂️[DNS API Dev Guide](https://github.com/acmesh-official/acme.sh/wiki/DNS-API-Dev-Guide)
